diff --git a/media/libmedia/MediaCodecBuffer.cpp b/media/libmedia/MediaCodecBuffer.cpp
index c35e0c6..68ae3ea 100644
--- a/media/libmedia/MediaCodecBuffer.cpp
+++ b/media/libmedia/MediaCodecBuffer.cpp
@@ -52,7 +52,8 @@ size_t MediaCodecBuffer::offset() const {
 }
 
 status_t MediaCodecBuffer::setRange(size_t offset, size_t size) {
-    return mBuffer->setRangeWithStatus(offset, size);
+    mBuffer->setRange(offset, size);
+    return OK;
 }
 
 sp<AMessage> MediaCodecBuffer::meta() {
