--- a/packages/SystemUI/src/com/android/systemui/shade/NotificationShadeWindowControllerImpl.java
+++ b/packages/SystemUI/src/com/android/systemui/shade/NotificationShadeWindowControllerImpl.java
@@ -489,13 +489,10 @@ public class NotificationShadeWindowControllerImpl implements NotificationShadeW
         boolean dreamShowingAndRotationAllowed = dreamsV2() && mContext.getResources().getBoolean(
                 com.android.internal.R.bool.config_alwaysAllowDreamRotation)
                 && state.isOnOrGoingToDream;
-        if (state.bouncerShowing || (state.isKeyguardShowingAndNotOccluded()
-                && !dreamShowingAndRotationAllowed) || state.dozing) {
-            if (shouldEnableKeyguardScreenRotation()) {
-                mLpChanged.screenOrientation = ActivityInfo.SCREEN_ORIENTATION_USER;
-            } else {
-                mLpChanged.screenOrientation = ActivityInfo.SCREEN_ORIENTATION_NOSENSOR;
-            }
+        if ((state.bouncerShowing || (state.isKeyguardShowingAndNotOccluded()
+                && !dreamShowingAndRotationAllowed) || state.dozing)
+                && shouldEnableKeyguardScreenRotation()) {
+            mLpChanged.screenOrientation = ActivityInfo.SCREEN_ORIENTATION_USER;
         } else if (state.glanceableHubOrientationAware || dreamShowingAndRotationAllowed) {
             mLpChanged.screenOrientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR;
         } else {
-- 
