diff --git a/services/core/java/com/android/server/policy/WindowWakeUpPolicy.java b/services/core/java/com/android/server/policy/WindowWakeUpPolicy.java
--- a/services/core/java/com/android/server/policy/WindowWakeUpPolicy.java
+++ b/services/core/java/com/android/server/policy/WindowWakeUpPolicy.java
@@ -25,13 +25,16 @@ import static android.os.PowerManager.WAKE_REASON_WAKE_MOTION;
 import static android.view.KeyEvent.KEYCODE_POWER;
 
 import static com.android.server.policy.Flags.supportInputWakeupDelegate;
+import static com.android.server.policy.WindowManagerPolicy.WindowManagerFuncs.LID_BEHAVIOR_NONE;
 import static com.android.server.power.feature.flags.Flags.perDisplayWakeByTouch;
 
 import android.annotation.Nullable;
+import android.content.ContentResolver;
 import android.content.Context;
 import android.os.PowerManager;
 import android.os.PowerManager.WakeReason;
 import android.os.SystemClock;
+import android.provider.Settings;
 import android.util.Slog;
 import android.view.Display;
 import android.view.KeyEvent;
@@ -44,6 +47,7 @@ import com.android.server.LocalServices;
 class WindowWakeUpPolicy {
     private static final String TAG = "WindowWakeUpPolicy";
 
+    private final Context mContext;
     private final PowerManager mPowerManager;
     private final Clock mClock;
 
@@ -56,6 +60,7 @@ class WindowWakeUpPolicy {
 
     @VisibleForTesting
     WindowWakeUpPolicy(Context context, Clock clock) {
+        mContext = context;
         mPowerManager = context.getSystemService(PowerManager.class);
         mClock = clock;
 
@@ -170,7 +175,11 @@ class WindowWakeUpPolicy {
      *      executed; {@code false} otherwise.
      */
     boolean wakeUpFromLid() {
-        wakeUp(mClock.uptimeMillis(), WAKE_REASON_LID, "LID");
+        int lidBehavior = Settings.Global.getInt(mContext.getContentResolver(),
+                Settings.Global.LID_BEHAVIOR, LID_BEHAVIOR_NONE);
+        if (lidBehavior != LID_BEHAVIOR_NONE) {
+            wakeUp(mClock.uptimeMillis(), WAKE_REASON_LID, "LID");
+        }
         return true;
     }
 
-- 

